packages/vue-app/template/server.js

```diff
@@ -321,8 +321,7 @@ const render404Page = () => {
         promise.then((asyncDataResult) => {
           ssrContext.asyncData[Component.cid] = asyncDataResult
           applyAsyncData(Component)
-          return asyncDataResult
-        })
+        }).catch(() => {})
```

Comment:
After discussing this a bit with @danielroe, I realized then maybe the intention here was to push this chained promise to `promises` rather than the original one? So instead of the current code do:

```js
        let promise = promisify(Component.options.asyncData, app.context)
        promise = promise.then((asyncDataResult) => {
          ssrContext.asyncData[Component.cid] = asyncDataResult
          applyAsyncData(Component)
          return asyncDataResult
        })
        promises.push(promise)

```

(with or without extra `.catch()`).

But that's more risky change as crash in `.then` handler would then affect other code up the chain. Maybe that would be the right thing to do but the change I made is at least safe from the perspective of not changing the behavior apart from catching unhandled rejection.

Reply:
Yeah, I think that's a better fix. And it seems quite safe to me as `applyAsyncData` is very simple.